contracts/abstracted_account.algo.ts

```diff
@@ -18,6 +28,15 @@ export class AbstractedAccount extends Contract {
    */
   namedPlugins = BoxMap<bytes, Application>({ prefix: 'n' });
 
+  /**
+   * Gates the ability to use plugins via a time lock
+   * & box key `appID` + `address
+   * GateParams is a tuple that contains the 
+   * duration of the gate and the time it was added
+   */
+  gates = BoxMap<bytes, GateParams>({ prefix: 'g' });
```

Comment:
Rather than having a separate gates box I think it'd be better to have the plugins key to be app + address and the value to be the end time. This reduces the required box references and reads.

Reply:
Wholeheartedly agree, whatever saves us space and still allows us to work the same functionally without losing any data.